src/main/app/auth/user/oidc.ts

```diff
@@ -39,17 +43,24 @@ export const getUserDetails = async (
 };
 
 export const getSystemUser = async (): Promise<UserDetails> => {
-  const id: string = config.get('services.idam.clientID');
-  const secret: string = config.get('services.idam.clientSecret');
-  const tokenUrl: string = config.get('services.idam.tokenURL');
-
   const systemUsername: string = config.get('services.idam.systemUsername');
   const systemPassword: string = config.get('services.idam.systemPassword');
 
-  const headers = { Accept: 'application/json', 'Content-Type': 'application/x-www-form-urlencoded' };
-  const data = `grant_type=password&username=${systemUsername}&password=${systemPassword}&client_id=${id}&client_secret=${secret}&scope=openid%20profile%20roles%20openid%20roles%20profile`;
+  let response;
+  const isNotProd: boolean = process.env.NODE_ENV !== 'production';
+  if (idamTokenCache.get(systemUsername) && isNotProd) {
+    logger.info('Fetching systemUsername from cache...');
+    response = idamTokenCache.get(systemUsername);
+  } else if (isNotProd) {
+    logger.info('Generating access token for systemUsername...');
+    response = await IdamUserManager.getAccessTokenFromIdam(systemUsername, systemPassword);
+    idamTokenCache.set(systemUsername, {
+      data: { id_token: response.data.id_token, access_token: response.data.access_token },
+    });
```

Comment:
do we know how it takes for the idam token to expire, I assume the cache would need to use the same expiry duration. also, do we need a dependency for this, could we not just store token in-session ourselves?